### ISSUE TITLE
Question regarding license of provided data

### ISSUE THREAD

**Issue by andaryjo** [public-transport/transport.rest#4]
I'm wondering under what license the data of the transport.rest APIs is provided - if any. I suspect that using their internal Hafas endpoints isn't really appreciated by the public transport companies, especially as all of the companies listed here also provide public APIs where the consumer has to authenticate himself and has to agree to terms of use.

Personally, I made a bad experience with using data from a public transport provider's internal interface and publishing it via my own app without their permission - up to the point where they were threatening me with legal steps. So I guess distributing Hafas client software for people to use is fine, but distributing data to the public via a self-hosted service seems risky to me.

Would it be safe to use the data for public projects?

**Comment by derhuerst** [comment]
Glad that you ask! I think documenting this would be useful for people trying to decide if they want to use the API(s).

Every once in a while, I've been in touch with some officials (or people working with the relevant officials) about both the underlying HAFAS client, but also about the public wrapper APIs. **I have received mixed feedback** about them, ranging from "I appreciate what you're doing, I think it's for everyone's benefit, so I personally won't get you into legal trouble, even though technically it's not allowed" to "I do not approve of what you're doing there, please stop" (see also https://github.com/derhuerst/vbb-rest/issues/29#issuecomment-547017864), but no further legal steps.

I'm speculating, but the reason why I think this project is being tolerated: The transit providers paying for the operation of these APIs (VBB, BVG, HVV, etc) desperately try to keep up with the fast-pacing digitalisation of urban mobility, but they hardly manage. Everyone is talking about inter-connected & intelligent transportation systems and intermodal routing, and transit operators know that they need to open up data access to stay an attractive mobility option; At the same time, from what I've heard, the traditional tech companies selling IT systems like HAFAS (obviously) don't offer as open and developer-friendly products as the transit providers would want, or the transit providers don't have the negotiation power or skills to put them in place. Again, this is my subjective informed guess, but they will tolerate devs like me as long as they're not *forced* to intervene, either because of legal reasons (because some data owner might complain, or the company selling HAFAS might complain that this "is contract-violating usage") of financial pressure (e.g. because the transit providers pay per HAFAS request lol).

So far, being transparent about these projects (e.g. by talking to them, and by sending a helpful `User-Agent` to the underlying HAFAS APIs) **has not caused any legal trouble** (for me or anyone I've talked to about this). **In case it ever does, I will probably shut off the relevant `transport.rest` APIs immediately.**

**TLDR: There's no open license for the data exposed by `*.transport.rest` APIs, and I don't have an official & explicit approval**, but they seem to be tolerated or flying under the data; So far, this has worked very well, and enabled many projects, from small proof-of-concepts (as part of their official hackathons!) to medium-scale semi-production apps. Also, [IANAL](https://en.wikipedia.org/wiki/IANAL), but I don't see how *consuming* the data might get *you* in trouble.

**Comment by derhuerst** [comment]
Does that answer your questions?

**Comment by derhuerst** [comment]
@andaryjo BTW: There are some other stories about trouble using transit data inofficially in Germany: [openPlanB](https://blog.wikimedia.de/2012/09/28/deutsche-bahn-openplanb-offener-brief/) & [Öffi app](https://play.google.com/store/apps/details?id=de.schildbach.oeffi) has been forced to take some API integrations offline in the past.

**Comment by andaryjo** [comment]
Thanks you so much for the detailed response.

I think this is a delicate topic as I can somewhat understand both sides. As a developer I'd like to have unlimited, standardized and easy access to public transport data, but a data provider probably has to think about SLAs and its own liability in legal conflicts. I can see why they would want to make sure to know the data consumers and to get them to sign a contract.

I'm actually surprised on how many public transport companies in Germany offer public APIs (not that it's a good rate, but still... I expected less), but it's a shame that many of them lack basic features and that there is still no real consolidated, standardized API design.

The only thing that comes close to the wide-spread distribution of Hafas is the TRIAS specification, which is used by 8 of the providers I work with (with VRR and MVV recently joining the ship) and allows me to realize somethin close to standardized data platform where all I really have to do is switching the provider's URL. TRIAS is still a bit edgy, but has all the relevant functionalities and (and that is important in my opinion) actually allows developers to use licensed data in their public projects without needing to worry about upsetting the public transport companies.

However, we still have a long way to go to achieve real open data...

**Comment by derhuerst** [comment]
> I'm actually surprised on how many public transport companies in Germany offer public APIs (not that it's a good rate, but still... I expected less), but it's a shame that many of them lack basic features and that there is still no real consolidated, standardized API design.

Almost all of these APIs are public, but not open: They require a contract to be signed, with quite strict conditions about how one is allowed to use them.

> The only thing that comes close to the wide-spread distribution of HAFAS is the TRIAS specification, which is used by 8 of the providers I work with (with VRR and MVV recently joining the ship) and allows me to realize something close to standardized data platform where all I really have to do is switching the provider's URL.

Yeah, from the German perspective it looks exciting (at least *some* standardisation), but IMHO internationally it is still deemed to fail as a standard.

> TRIAS is still a bit edgy, but has all the relevant functionalities and (and that is important in my opinion) actually allows developers to use licensed data in their public projects without needing to worry about upsetting the public transport companies.

Do the TRIAS API endpoints out there by now usually come without restrictions, or at least with a very open legal contract/license?

I'm interested in details about TRIAS: Resources to learn about it, more details about the endpoints and their TRIAS support, open source software for anything TRIAS. Let's chat about it in another thread!

**Comment by andaryjo** [comment]
I'd like to move further dicussion on TRIAS here: https://github.com/public-transport/ideas/issues/18 :relaxed:
